python/ray/data/datasource/file_based_datasource.py

```diff
@@ -300,31 +316,25 @@ def write(
             open_stream_args = {}
 
         def write_block(write_path: str, block: Block):
-            logger.debug(f"Writing {write_path} file.")
+            logger.get_logger().debug(f"Writing {write_path} file.")
             fs = _unwrap_s3_serialization_workaround(filesystem)
             if _block_udf is not None:
                 block = _block_udf(block)
 
-            with fs.open_output_stream(write_path, **open_stream_args) as f:
-                _write_block_to_file(
-                    f,
-                    BlockAccessor.for_block(block),
-                    writer_args_fn=write_args_fn,
-                    **write_args,
-                )
+            block = BlockAccessor.for_block(block)
+            # Skip write if the block is empty.
+            if block.num_rows() > 0:
```

Comment:
num_rows cannot be empty here, because we've already returned earlier, right? we can just do `assert num_rowws != 0` here.

Reply:
I think the block can still be empty here. the earlier return only occurs when all of the blocks are empty (we first combine all input blocks and build a single block, then check if the number of rows is zero). Therefore, we could still have a case where some blocks have rows, and some blocks are empty. Then in this individual `write_block` method, we would need to differentiate between writing blocks with rows versus skipping empty blocks.